extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -102,16 +104,26 @@ public JpaEntitiesBuildItem discoverModelAndRegisterForReflection() throws IOExc
             }
         }
 
+        if (!unindexedClasses.isEmpty()) {
+            final String unindexedClassesErrorMessage = unindexedClasses.stream().map(d -> "\t- " + d + "\n")
+                    .collect(Collectors.joining());
+            throw new ConfigurationError(
+                    "Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:\n"
+                            + unindexedClassesErrorMessage
+                            + "Consider adding them to the index either by creating a Jandex index " +
+                            "for your dependency or via quarkus.index-dependency properties.");
+        }
+
         return domainObjectCollector;
     }
 
     private static void enlistExplicitClasses(IndexView index, JpaEntitiesBuildItem domainObjectCollector,
-            Set<String> enumTypeCollector, List<String> managedClassNames) {
+            Set<String> enumTypeCollector, List<String> managedClassNames, Set<DotName> unindexedClasses) {
         for (String className : managedClassNames) {
             DotName dotName = DotName.createSimple(className);
             boolean isInIndex = index.getClassByName(dotName) != null;
             if (isInIndex) {
-                addClassHierarchyToReflectiveList(index, domainObjectCollector, enumTypeCollector, dotName);
+                addClassHierarchyToReflectiveList(index, domainObjectCollector, enumTypeCollector, dotName, unindexedClasses);
             } else {
                 // We do lipstick service by manually adding explicitly the <class> reference but not navigating the hierarchy
```

Comment:
Shouldn't we use the same pattern here and add the class to the `unindexedClasses`?

Reply:
@gsmet I was tempted to do so but the comment made it look like something else here is done: add the class anyway and do not scan the whole hierarchy.

Reply:
It is the same [here](https://github.com/quarkusio/quarkus/blob/6b5bc32175d08538523dca2196bc91bd8a726dae/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java#L133). 

I was also thinking if there is a way to document on how to  "index" custom dependencies as suggested by this [line](https://github.com/quarkusio/quarkus/blob/6b5bc32175d08538523dca2196bc91bd8a726dae/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java#L131) but with working link.